p4-16/spec/P4-16-spec.mdk

```diff
@@ -5276,8 +5266,8 @@ following behavior after the expression evaluation is interrupted.
 ## Conditional statement { #sec-cond-stmt }
 
 The conditional statement uses standard syntax and semantics familiar from many programming languages.
-However, the condition expression in P4 is required to be a Boolean (and not
-an integer). Conditional statements may not be used within a `parser`.
+However, the condition expression in P4 is required to be a Boolean (and not an integer).
+Conditional statements cannot be used within a `parser`.
```

Comment:
At one point I vaguely recall a discussion about adding a pass that will convert conditionals (or at least ifs) in parsers into additional states.  Regardless, this seems like something that should be architecture-dependent -- some architectures may have no problem with conditionals in parser while others may not be able to deal with them (or require converting them to additional states).  Did will ever add language about "may not" referring to architecture restrictions, while "shall not" (or something stronger) would refer to language restrictions?

I think there's some discussion of this in the architecture section, but probably not enough

Reply:
Conditionals are currently fully supported in the parser. They have been since P4 1.2.2, in fact.

Reply:
Interesting.  I didn`t know about this, or at least not _much_.  I was just trying to improve the use of English grammar.

Reply:
@ChrisDodd & @mbudiu-vmw : is it OK to resolve this "conversation"?